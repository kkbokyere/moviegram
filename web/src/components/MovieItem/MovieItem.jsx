import React from 'react';
import PropTypes from 'prop-types';

const MovieItem = ({ img, title, showTimes = [], id }) => {
    return (
        <article className="movie-item">
            <div>
            <img className="movie-item__img" src={img} alt={title}/>
            </div>
            <div className="movie-item__wrapper">
                <h1 className="movie-item__title">{title}</h1>
                <div className="movie-item__showtimes">
                    <h3>Show Times:</h3>
                    {showTimes.map(({time, cinema}, index) => <span className="movie-item__time" key={index}>{time} @ {cinema}</span>)}
                </div>
                <button className="btn btn-primary">See more</button>
            </div>
        </article>
    );
};

MovieItem.propTypes = {
    img: PropTypes.string,
    title: PropTypes.string,
    showTimes: PropTypes.array,
    id: PropTypes.string,
};

export default MovieItem;