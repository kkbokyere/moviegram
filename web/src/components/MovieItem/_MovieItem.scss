.movie-item {
  border-bottom: 1px solid $primary-grey;
  display: flex;
  margin-bottom: $default-padding;
  padding: $default-padding 0;
}

.movie-item__title {
}

.movie-item__img {
  margin-right: $default-padding;
}

.movie-item__showtimes {

  margin-bottom: $default-padding;
}

.movie-item__time {
  display: block;
  margin-bottom: 10px;
}