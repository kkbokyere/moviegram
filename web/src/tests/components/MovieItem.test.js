import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MovieItem from '../../components/MovieItem';

describe('MovieItem Component', () => {
    it('should render MovieItem', () => {
        const wrapper = renderer.create(<MovieItem />).toJSON();

        expect(wrapper).toMatchSnapshot()
    });

    it('should render MovieItem with data', () => {
        const movieItemData = {
            "id": "1",
            "title": "AVENGERS: ENDGAME",
            "description": "The grave course of events set in motion by Thanos that wiped out half the universe and fractured the Avengers ranks compels the remaining Avengers to take one final stand in Marvel Studios' grand conclusion to twenty-two films",
            "showTimes": [
                {"time": "May 16th 2019, 3:24:25 pm", "cinema":"Wood Green - Vue"},
                {"time": "May 16th 2019, 3:24:25 pm", "cinema": "Enfield - Cineworld"}
            ],
            "img": "https://via.placeholder.com/200x300",
            "imgWide": "https://via.placeholder.com/400x300"
        };

        const movieItemComponent = mount(<MovieItem {...movieItemData} />);

        expect(movieItemComponent.find('.movie-item__title').text()).toEqual(movieItemData.title)
    });

});