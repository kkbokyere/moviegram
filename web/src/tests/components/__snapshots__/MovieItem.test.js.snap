// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem Component should render MovieItem 1`] = `
<article
  className="movie-item"
>
  <div
    className="movie-item__img"
  >
    <picture>
      <source
        media="(min-width: 768px)"
      />
      <img />
    </picture>
  </div>
  <div
    className="movie-item__wrapper"
  >
    <h1
      className="movie-item__title"
    />
    <div
      className="movie-item__showtimes"
    >
      <h3>
        Show Times:
      </h3>
    </div>
    <a
      className="btn btn-primary"
      href="/details/undefined"
      onClick={[Function]}
    >
      See more
    </a>
  </div>
</article>
`;
